src/java/boa/types/BoaArray.java

```diff
@@ -59,6 +59,9 @@ public boolean assigns(final BoaType that) {
 			return true;
 		}
 
+		if (that instanceof BoaProtoList)
```

Comment:
This isnt sufficient.  We also want to trap assignment of the form `array of Expression` and assigning to it `protolist of Statement` but this current check allows assigning **any** protolist to **any** array.

Reply:
I there any other assignment that we want to trap? Or just `array of x` to `protolist of x` and `array of Expression` to `protolist of Statement`?